db/migrate/20250605133700_create_scim_clients.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class CreateScimClients < ActiveRecord::Migration[8.0]
+  def change
+    create_table :scim_clients do |t|
+      t.string :name, null: false
+      t.belongs_to :auth_provider, null: false
```

Comment:
I suggest adding db constraint as well.
```suggestion
      t.belongs_to :auth_provider, null: false, on_delete: :restrict
```

Reply:
Excellent point. I was under the impression that the foreign key would be created implicitly. Having a foreign key without anything specified, usually means restrict then.

However, I was wrong: No foreign key was created, so no database guarantees. Fixed.